p2p/swarm.go

```diff
@@ -178,7 +179,8 @@ func newSwarm(ctx context.Context, config config.Config, newNode bool, persist b
 
 	s.cPool = cpool
 
-	s.gossip = gossip.NewProtocol(config.SwarmConfig, s, s.LocalNode().PublicKey(), s.lNode.Log)
+	s.pq = gossip.NewPriorityQ(10) // TODO: set according to proto
```

Comment:
I don't see any advantages in the way you implemented this. partially as a part of gossip but partially not. 
It looks like you tried to decouple but eventually coupled in a weird way. I think we can live with a constant size of priorities. like 5 (HIGH, MID-HIGH, MID, MID-LOW, LOW) or even less like 3 (high,mid,low) to make things more simple.

Reply:
"I don't see any advantages in the way you implemented this. partially as a part of gossip but partially not." - what does it mean? what are the disadvantages you see if any?

"It looks like you tried to decouple but eventually coupled in a weird way." - what is coupled?

Elaborate on your comments, I have no use in a "buzz words" review. Be Specific so I can address your changes requests in any way.

Reply:
This impl was the quickest and was intended to work as a POC. See that it actually helps us.
It is true we need to think about what should be the real impl. That is why I suggested this PR. Easier to see and comment and even change the design if needed.
Having a predetermined set of priorities sounds good and enough for us, I'll go over it to decide how many we need.

Reply:
These ain't not buzzwords. the quickest isn't necessarily the best. what I see is that you created a dependency between `swarm` and gossip.`prioq` and a dependency between `prioq` and `gossip.Protocol` I'd rather the prio-q to be contained in gossip as it is only used there. there's no need for swarm to be aware of prioq and it complicates the `swarm` impl for no good reason.

Reply:
If priority is a feature of gossip, it should be interfaced using the gossip api.

Reply:
"the quickest isn't necessarily the best." - when did I say the quickest is the best? The quickest was our best move back then. Now, after we believe it works we can discuss what should be the solution or if any changes are required. That is exactly the purpose of the PR. Exactly as I stated before.